classes/activity-manager.js

```diff
@@ -119,9 +119,10 @@ class ActivityManager {
             const member = activity.generalText.guild.member(user);
 
             if (!seenUsers.has(user.id)) {
-                const regex = RegExp(/^.+\s#\d{1,2}$/);
+                // const regex = RegExp(/^.+\s#\d{1,2}$/);
 
-                let role = member.roles.cache.find(role => regex.test(role.name));
+                // let role = member.roles.cache.find(role => regex.test(role.name));
+                let role = member.roles.cache.find(role => discordServices.stampRoles.has(role.id));
 
                 if (role != undefined) this.parseRole(member, role, activity.name);
```

Comment:
Could we possibly give the user the 0th stamp if there are no stamps found? @mwang2000

Reply:
We could, only thing is that if a non-hacker (sponsor, mentor, staff) decides to click on it they will get a stamp role. We could restrict it so that people with those roles won't get the 0th stamp. Also, would it make more sense to give them the 1st stamp when they click on it?

Reply:
update: added a TODO here as well because a reference to the 0th stamp is required @JPGarCar